scripts/test/system.sh

```diff
@@ -1,6 +1,6 @@
 #!/usr/bin/env bash
 
-set -em -o pipefail
```

Comment:
The problem is that `_subctl` ignored errors previously anyway. Fixing `_subctl` so that it no longer ignores errors, which I need so that the tests can actually verify that certain forms of `subctl` fail, causes the overall system test to fail earlier because `subctl` exits with an error in certain “normal” cases anyway.

Since `set -e -o pipefail` was effectively meaningless, I removed it; we can add it back later, once we have a consistent set of rules for `subctl`’s exit code.

Reply:
I don't understand what you mean by
> `_subctl` ignored errors previously anyway

and
> Since `set -e -o pipefail` was effectively meaningless

It didn't ignore errors, it would fail the tests due to any error, eg try to apply this and run the system tests:

```patch
diff --git a/scripts/test/system.sh b/scripts/test/system.sh
index 69a4e9c..62d470d 100755
--- a/scripts/test/system.sh
+++ b/scripts/test/system.sh
@@ -133,7 +133,7 @@ deploy_env_once
 
 # Test subctl show invocations
 
-_subctl show all | tee /dev/stderr | (sponge ||:) | grep -q 'Cluster "cluster2"'
+_subctl show all --sdfsdf | tee /dev/stderr | (sponge ||:) | grep -q 'Cluster "cluster2"'
 # Single-context variants don't say 'Cluster "foo"', check what cluster is considered local
 _subctl show all --context cluster1 | tee /dev/stderr | (sponge ||:) | grep -qv 'cluster2.*local'
 _subctl show all --context cluster2 | tee /dev/stderr | (sponge ||:) | grep -q 'cluster2.*local'
```

Reply:
Yes that one is indeed problematic but it doesn't have anything to do with the `set -e`..

You can make this change to ensure it (as we don't care about the specific exit code):
```suggestion
    "${DAPPER_SOURCE}"/cmd/bin/subctl "$@" || exit 1
```

GHA will be fine without a closing `::endgroup::` anyway, and the code will be tidier :)